packages/calling/src/CallingClient/calling/call.test.ts

```diff
@@ -1955,6 +1955,191 @@ describe('State Machine handler tests', () => {
+      expect(setTimeoutForStateSpy).toHaveBeenCalledWith(20000);
+      expect(clearCallEventsTimeoutSpy).toHaveBeenCalledTimes(1);
+
+      // Media setup for the call
+      dummyEvent.type = 'E_SEND_ROAP_OFFER';
+      call.sendMediaStateMachineEvt(dummyEvent as RoapEvent);
+
+      dummyEvent.data = {
+        seq: 1,
+        messageType: 'OFFER',
+        sdp: 'sdp',
+      };
+      call.sendMediaStateMachineEvt(dummyEvent as RoapEvent);
+
+      dummyEvent.type = 'E_RECV_ROAP_ANSWER';
+      call.sendMediaStateMachineEvt(dummyEvent as RoapEvent);
+
+      const dummyOkEvent = {
+        type: 'E_ROAP_OK',
+        data: {
+          received: false,
+          message: {
+            seq: 1,
+            messageType: 'OK',
+          },
+        },
+      };
+      call.sendMediaStateMachineEvt(dummyOkEvent as RoapEvent);
+      dummyEvent.type = 'E_RECV_ROAP_OFFER_REQUEST';
+      call.sendMediaStateMachineEvt(dummyEvent as RoapEvent);
+      dummyEvent.type = 'E_SEND_ROAP_OFFER';
+      call.sendMediaStateMachineEvt(dummyEvent as RoapEvent);
+      dummyEvent.type = 'E_RECV_ROAP_ANSWER';
+      call.sendMediaStateMachineEvt(dummyEvent as RoapEvent);
+      call.sendMediaStateMachineEvt(dummyOkEvent as RoapEvent);
+
+      setTimeoutForStateSpy.mockClear();
+      clearCallEventsTimeoutSpy.mockClear();
+
+      dummyEvent.type = 'E_SEND_CALL_CONNECT';
+      await call.sendCallStateMachineEvt(dummyEvent as CallEvent);
+      expect(setTimeoutForStateSpy).toHaveBeenCalledWith(60000);
+
+      expect(call['callStateMachine'].state.value).toBe('S_CALL_ESTABLISHED');
+      expect(call.isConnected()).toBe(true);
+    });
+
+    it('times out if the next event is not received', async () => {
+      const statusPayload = <WebexRequestPayload>(<unknown>{
+        statusCode: 200,
+        body: mockStatusBody,
+      });
+      const dummyEvent = {
+        type: 'E_SEND_CALL_SETUP',
+        data: undefined as any,
+      };
+      callManager.callCollection = {};
+      const call = callManager.createCall(dest, CallDirection.OUTBOUND, deviceId, mockLineId);
+      const emitSpy = jest.spyOn(call, 'emit');
+      const deleteSpy = jest.spyOn(call as any, 'delete');
+      const logSpy = jest.spyOn(log, 'warn');
+      webex.request.mockReturnValue(statusPayload);
+      expect(Object.keys(callManager.callCollection)[0]).toBe(call.getCorrelationId());
+
+      // handleOutgoingCallSetup is asynchronous
+      await call.sendCallStateMachineEvt(dummyEvent as CallEvent);
+      expect(call['callStateMachine'].state.value).toBe('S_SEND_CALL_SETUP');
+      logSpy.mockClear();
+      jest.advanceTimersByTime(70000);
+      expect(logSpy.mock.calls[0][0]).toBe('Call timed out in state: S_SEND_CALL_SETUP');
+      expect(emitSpy).toHaveBeenCalledWith(CALL_EVENT_KEYS.DISCONNECT, call.getCorrelationId());
+      expect(deleteSpy).toHaveBeenCalledTimes(1);
+      // @ts-ignore
+      expect(Object.keys(callManager.callCollection)[0]).toBe(undefined);
```

Comment:
can we not check for the collection to be empty?

Reply:
Updated